### DISCUSSION TITLE
Validation breaking change

### DISCUSSION THREAD

**Discussion by ju5t** [livewire/livewire#5853] [1. Help]
v3 is 🚀 

Our app uses dynamic validation in a few places. This means we dynamically create our validation when `rules()` is called, alongside with their corresponding validation messages in `messages()`. It appears that `rules()` works when you're using forms, but messages aren't. Am I missing something or is this not supported anymore?

**Comment by joshhanley** [comment]
@ju5t `messages()` should still work too. Are they not working for you?

https://github.com/livewire/livewire/blob/04298292a365841879d64c13881d5499517bb597/src/Features/SupportValidation/HandlesValidation.php#L109-L117

**Reply by ju5t** [reply]
With `messages` inside the form object it doesn't work for me. When I move the `messages` inside the component like in v2, my custom messages do work. Validation with `rules` works when its in the form and also when its in the component.

This is my form object.

```php
public string $content = '';

public function rules(): array
{
    return [
        'content' => 'required',
    ];
}

public function messages(): array
{
    return [
        'data.content.required' => 'We need at least something!',
    ];
}
```

I've tried changing the `messages` array both with and without the `data`-prefix. But neither worked for me. The form is linked to our component like so:

```php
public PostForm $data;
```

**Reply by joshhanley** [reply]
@ju5t ah sorry, I missed the part where you said you were using FormObjects. Yeah maybe that isn't supported yet. PRs are welcome! 🙂

**Reply by viliusvsx** [reply]
Added PR for dynamic validation attributes using from Form Object [#5904](https://github.com/livewire/livewire/pull/5904)

**Reply by joshhanley** [reply]
@viliusvsx thanks!

**Reply by ju5t** [reply]
Cool! Thanks @viliusvsx!
